week1/homework/src/server.js

```diff
@@ -1,20 +1,42 @@
 'use strict';
 
-const http = require('http');
+{
+  const http = require('http');
 
-/* `createServer` MUST return an instance of `http.Server` otherwise the tests
- * will fail.
- */
-function createServer(port) {
-  let state = 10;
+  const postData = (status, response, data) => {
+    response.writeHead(status, {
+      'Content-Type': 'application/json'
+    });
+    response.end(JSON.stringify(data));
+  };
 
-  const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
-  });
+  function createServer(port) {
+    let state = 10;
 
-  return server;
-}
+    const server = http.createServer((request, response) => {
+      switch (request.url) {
+        case '/state':
+          postData(200, response, { state: state });
+          break;
+        case '/add':
+          postData(200, response, { state: ++state });
+          break;
+        case '/subtract':
+          postData(200, response, { state: --state });
+          break;
+        case '/reset':
+          postData(200, response, { state: (state = 10) });
```

Comment:
interesting approach. Keep in mind that in other languages assignment operation does not always return the same result. For example in c I believe the result of an assignment is always 1.  Sometimes using such clever tricks can fire back and lead to a very painful debugging session. So best approach would be to keep things explicit even if that means writing more lines of code.